### PR TITLE
Improved docker guide and installation recommendation (related to #812)

### PR DESCRIPTION
**Pull request by berndruecker**

### CODE REVIEW

**Comment by akeller** [comment]
@berndruecker let us know if you need help with links 😄

**Comment by berndruecker** [comment]
@akeller ah sorry - fixed the link now. Good to go from my end !

**Review by christinaausley — DISMISSED** [review]
A few minor adjustments on product naming in my latest commits for uniformity purposes (Camunda Platform 8, Self-Managed) as well as a few typo corrections.

Had just a few comments which could improve this set of docs -- let me know your thoughts.

File: docs/self-managed/platform-deployment/docker.md [hunk 1]
```diff
@@ -3,254 +3,79 @@ id: docker
...
+| Zeebe     | [camunda/zeebe:latest](https://hub.docker.com/r/camunda/zeebe) | [Environment variables](../../zeebe-deployment/configuration/environment-variables/) |
+| Operate   | [camunda/operate:latest](https://hub.docker.com/r/camunda/operate) | [Operate configuration](../../operate-deployment/configuration) |
+| Tasklist  | [camunda/tasklist:latest](https://hub.docker.com/r/camunda/tasklist) | [Tasklist configuration](../../tasklist-deployment/configuration) |
+| Identity  | [camunda/identity:latest](https://hub.docker.com/r/camunda/identity) | [Configuration variables](../../identity/deployment/configuration-variables/) |
+| Optimize  | [camunda/optimize:latest](https://hub.docker.com/r/camunda/optimize) | [Environment variables](../../optimize-deployment/install-and-start/#available-environment-variables) |
 
-You can run Zeebe with Docker:
+Zeebe is the only component that is often run without any of the others. In this scenario, it does not need anything else, so a simple `docker run` is sufficient:
```
[the remark this answers is not in the document]
Comment (berndruecker) [reply docs/self-managed/platform-deployment/docker.md:24]: What I meant was, that "Zeebe" is the only component that make sense to run standalone. E.g. Operate make no sense to run without Zeebe.

File: docs/self-managed/platform-deployment/docker.md [hunk 2]
```diff
@@ -3,254 +3,79 @@ id: docker
...
+## Docker Compose
 
-You can use the Docker image `camunda/tasklist:latest` to run Tasklist as a container.
+A Docker Compose configuration to run Zeebe, Operate, Tasklist, Optimize, and Identity is available in the [camunda-platform](https://github.com/camunda/camunda-platform/blob/main/docker-compose.yaml) repository.
+Follow the instructions in the [README](https://github.com/camunda/camunda-platform#using-docker-compose).
 
-:::note
-Configure the appropriate settings described in the [Tasklist configuration](../../tasklist-deployment/configuration) section.
+:::warning
+While the Docker images themselves are supported for production usage, the Docker Compose files are designed to be used by developers to run an environment locally; they are not designed to be used in production. We recommend to use [Kubernetes](../kubernetes) in production, see also [Installation Overview](./).
```
[the remark this answers is not in the document]
Comment (berndruecker) [reply docs/self-managed/platform-deployment/docker.md:42]: It should link to https://docs.camunda.io/docs/self-managed/platform-deployment/ - which this link should do (no?)

File: docs/self-managed/platform-deployment/index.md [hunk 3]
```diff
@@ -26,18 +26,39 @@ For details on supported environments (e.g. Java or Elasticsearch versions), see
...
+* [**Local installation**](./local): You can run the Java applications on a local or virtual machine if it provides a supported Java Virtual Machine (JVM). This allows you to run Camunda on virtual machines or bare metal and offers a significant amount of flexibility. However, you will need to configure the details for the components to interact correctly yourself. We consider this a last resort. Note that Windows/Mac is **not** supported for production usage of Zeebe.
 
-We strongly recommend
+## Deployment recommendation
 
-* For **production**: Use Kubernetes and our [Helm charts](./kubernetes-helm). This setup provides you with predictable and consistent configuration, and the ability to manage deployment using automation tools.
-* For **development**: Use SaaS. If this is not possible, use Docker Compose or Kubernetes locally. If this does not work either, consider starting a Zeebe broker via Java.
```
[the remark this answers is not in the document]
Comment (berndruecker) [reply docs/self-managed/platform-deployment/index.md:36]: Maybe this is closest: https://camunda.com/get-started

File: docs/self-managed/platform-deployment/docker.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (christinaausley) [reply docs/self-managed/platform-deployment/docker.md:24]: Thank you for clarifying here! Adjusted this in my latest commit for clarification.

File: docs/self-managed/platform-deployment/docker.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (christinaausley) [reply docs/self-managed/platform-deployment/docker.md:42]: Ah I see -- thanks for confirming.

File: docs/self-managed/platform-deployment/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (christinaausley) [reply docs/self-managed/platform-deployment/index.md:36]: Agreed here. Adjusted in latest commit.

**Review by christinaausley — APPROVED** [review]

**Pull request merged**